frame/staking/src/pallet/mod.rs

```diff
@@ -1769,6 +1801,41 @@ pub mod pallet {
 			MinCommission::<T>::put(new);
 			Ok(())
 		}
+
+		/// Migrates an account's `RewardDestination` in `Payee` to `PayoutDestination` in `Payees`
+		/// if a record exists and if it has not already been migrated.
+		///
+		/// Effects will be felt instantly (as soon as this function is completed successfully).
+		///
+		/// This will waive the transaction fee if the payee is successfully migrated.
+		///
+		/// ## Complexity
+		/// - O(1)
+		/// - Independent of the arguments. Insignificant complexity.
+		/// - Contains a limited number of reads.
+		/// - Writes are limited to the `who` account key.
+		#[pallet::call_index(26)]
+		#[pallet::weight(T::WeightInfo::update_payee())]
+		pub fn update_payee(
+			origin: OriginFor<T>,
+			controller: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
+			let stash = &ledger.stash;
+
+			// If this stash has already been migrated, return early and charge tx fees.
+			if Payees::<T>::contains_key(&stash) && !Payee::<T>::contains_key(&stash) {
+				return Ok(Pays::Yes.into())
+			}
+
+			Payees::<T>::insert(
+				stash.clone(),
+				Payee::<T>::get(&stash).to_payout_destination(stash.clone(), controller),
+			);
+
```

Comment:
Shouldn't we do `Payee::<T>::remove(&stash)` here as well? otherwise the check above will always fail and the payee entry linger until the stash is removed (although `kill_stash` is not removing the payee entry, which I think it should - see comment above).

Reply:
Ah yes, quite a major piece of missing logic there. Thanks for pointing that out!